modules/@apostrophecms/area/ui/apos/apps/AposAreas.js

```diff
@@ -3,27 +3,38 @@ import { klona } from 'klona';
 
 export default function() {
 
+  let widgetsRendering = 0;
+
   createWidgetClipboardApp();
 
   prepareAreas();
 
   document.documentElement.style.setProperty('--a-widget-margin', apos.ui.widgetMargin);
 
+  apos.bus.$on('widget-rendering', function() {
+    widgetsRendering++;
+  });
+
   apos.bus.$on('widget-rendered', function() {
-    prepareAreas();
+    widgetsRendering--;
+    prepareAreasAndRunPlayersWhenDone();
   });
+
   apos.bus.$on('refreshed', function() {
-    prepareAreas();
+    prepareAreasAndRunPlayersWhenDone();
   });
 
+  function prepareAreasAndRunPlayersWhenDone() {
+    prepareAreas();
+    if (widgetsRendering === 0) {
+      apos.util.runPlayers();
+    }
+  }
+
   function prepareAreas() {
```

Comment:
this fn is now just a wrapper to run `createAreaApps` .. can we just run `createAreaApps` directly inside `prepareAreasAndRunPlayersWhenDone` to reduce this scaffolding?